gdcmodels/force_merge_segements.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import argparse
+import threading
+import urllib3
+
+from itertools import cycle
+from time import sleep, time
+
+from elasticsearch import Elasticsearch
+from elasticsearch.exceptions import ConnectionTimeout
+
+
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        description="Force a merge on the shards of one or more ES indices"
+    )
+    parser.add_argument(
+        "--verify-certs",
+        default=False,
+        action="store_true",
+        help="Verify the ca certs for Elasticsearch",
+    )
+    parser.add_argument(
+        "--index",
+        help="The index to do the merge, if not provide, all indices will be merged.",
+    )
+    parser.add_argument(
+        "--host", required=True, help="Elasticsearch server host name or IP",
+    )
+    parser.add_argument(
+        "--port", default=9200, help="Elasticsearch server port (default: 9200)",
+    )
+    parser.add_argument("--user", default="", help="Elasticsearch client user")
+    parser.add_argument(
+        "--password", dest="password", default="", help="Elasticsearch client password"
+    )
+
+    return parser
+
+
+def _force_merge_wrapper(args):
+    es = Elasticsearch(
+        hosts=[{"host": args.host, "port": args.port}],
+        use_ssl=True,
+        verify_certs=args.verify_certs,
+        http_auth=(args.user, args.password),
+    )
+
+    return force_merge_elasticsearch_segments(es, args.index)
+
+
+def _force_merge(es, index):
```

Comment:
I don't think we should allow for `index` to be an empty string or `None`. This should result in a validation error

Reply:
How does the script decide when to merge on all indices?

https://elasticsearch-py.readthedocs.io/en/master/api.html#elasticsearch.client.IndicesClient.forcemerge
> `forcemerge(index=None, params=None, headers=None)`

> index – A comma-separated list of index names; use _all or empty string to perform the operation on all indices

Reply:
I don't think a default to all is a good default. I'm proposing a contract where you MUST specify a set of indices